lbrynet/wallet/manager.py

```diff
@@ -54,19 +55,65 @@ def network(self):
 
     @property
     def use_encryption(self):
-        # TODO: implement this
-        return False
+        return self.default_account.serialize_encrypted
 
     @property
     def is_first_run(self):
         return True
 
     @property
     def is_wallet_unlocked(self):
-        return True
+        return not self.default_account.encrypted
 
     def check_locked(self):
-        return defer.succeed(False)
+        return defer.succeed(self.default_account.encrypted)
+
+    def get_account_or_error(self, account_id: str, lbc_only=True):
+        if not account_id:
+            if lbc_only and not isinstance(self.default_account, LBCAccount):
+                raise ValueError(
+                    "Found '{}', but it's an {} ledger account. A LBC Account is needed."
+                        .format(account_id, self.default_account.ledger.symbol)
+                )
+            return self.default_account
+        for account in self.default_wallet.accounts:
+            if account.id == account_id:
+                if lbc_only and not isinstance(account, LBCAccount):
+                    raise ValueError(
+                        "Found '{}', but it's an {} ledger account. A LBC Account is needed."
+                        .format(account_id, account.ledger.symbol)
+                    )
+                return account
+        raise ValueError("Couldn't find account: {}.".format(account_id))
+
+    def decrypt_account(self, account_id):
+        account = self.get_account_or_error(account_id, lbc_only=False)
+        assert account.password is not None, "account is not unlocked"
+        assert not account.encrypted, "account is not unlocked"
+        account.serialize_encrypted = False
+        self.save()
+        return defer.succeed(not account.encrypted and not account.serialize_encrypted)
+
+    def encrypt_account(self, password, account_id):
+        account = self.get_account_or_error(account_id, lbc_only=False)
+        assert not account.encrypted, "account is already encrypted"
+        account.encrypt(password)
+        account.serialize_encrypted = True
+        self.save()
+        return defer.succeed(account.encrypted and account.serialize_encrypted)
+
+    def unlock_account(self, password, account_id):
+        account = self.get_account_or_error(account_id, lbc_only=False)
+        assert account.encrypted, "account is not locked"
+        account.decrypt(password)
+        return defer.succeed(not account.encrypted)
+
+    def lock_account(self, account_id):
+        account = self.get_account_or_error(account_id, lbc_only=False)
+        assert account.password is not None, "account is already locked"
+        assert not account.encrypted and account.serialize_encrypted, "account is not encrypted"
+        account.encrypt(account.password)
+        return defer.succeed(account.encrypted)
```

Comment:
The 4 methods above should all accept an account instance instead of an `account_id`. Also, can you make them return plain values instead of Deferreds, none of them are actually async.